Problem: block gas meter not compatible with parallel tx execution

yihuang opened the pull request:
## Description

Disable block gas meter, it's safe to do because we checks gas wanted against block gas limit in ProcessProposal now.

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

yihuang commented on CHANGELOG.md:
```diff
@@ -49,6 +49,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 ### State Machine Breaking
 
 * (rpc) [#443](https://github.com/crypto-org-chain/ethermint/pull/443) Keep behavior of random opcode as before.
+* (baseapp) [#]() Disable block gas meter, it's not compatible with parallel tx execution.
```
```suggestion
* (app) [#451](https://github.com/crypto-org-chain/ethermint/pull/451) Disable block gas meter, it's not compatible with parallel tx execution.
```

yihuang commented on CHANGELOG.md:
```diff
@@ -49,6 +49,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 ### State Machine Breaking
 
 * (rpc) [#443](https://github.com/crypto-org-chain/ethermint/pull/443) Keep behavior of random opcode as before.
+* (app) [#451](https://github.com/crypto-org-chain/ethermint/pull/451) Disable block gas meter, it's not compatible with parallel tx execution.
```
```suggestion
* (app) [#451](https://github.com/crypto-org-chain/ethermint/pull/451) Disable block gas meter, it's not compatible with parallel tx execution. It's safe to do as long as we checks total gas-wanted against block gas limit in process proposal, which we do in default handler.
```

mmsqe commented on x/evm/keeper/state_transition.go:
```diff
@@ -210,32 +210,22 @@ func (k *Keeper) ApplyTransaction(ctx sdk.Context, msgEth *types.MsgEthereumTx)
 		bloomReceipt = ethtypes.BytesToBloom(bloom.Bytes())
 	}
 
-	cumulativeGasUsed := res.GasUsed
```
seems we also dont use cumulativeGasUsed in [PostTxProcessing](https://github.com/crypto-org-chain/cronos/blob/main/x/cronos/keeper/evm_hooks.go#L48)

yihuang replied:
yeah, I bet no one would need this in consensus.

mmsqe reviewed (commented):
seems base fee need some adjustment after params change now

yihuang commented:
> seems base fee need some adjustment after params change now

maybe sth wrong with the gas accumulation, the `update_feemarket_param` test case is flacky though, when we checks the parameter again, it might be updated already.

Commit:
Problem: block gas meter not compatible with parallel tx execution

Disable block gas meter, it's safe to do because we checks gas wanted against block gas limit in ProcessProposal now.

Update CHANGELOG.md

Signed-off-by: yihuang <huang@crypto.com>

Update CHANGELOG.md

Signed-off-by: yihuang <huang@crypto.com>

cleanup

format

use block gas wanted in context

update deps

fix unit test

fix unit test

remove feemarket transient store

remove

update deps
